WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -147,13 +153,19 @@ final class DashboardViewModel: ObservableObject {
     /// The visibility is updated on `onAppear()` to consider scenarios when the app is
     /// never terminated.
     ///
-    func onViewAppear() {
+    @MainActor
+    func onViewAppear() async {
         refreshIsInAppFeedbackCardVisibleValue()
+
+        /// When the user creates a Blaze campaign after hiding the Blaze card
+        /// we add the Blaze card back in `BlazeCampaignCreationCoordinator`.
+        /// Here we need to get the updated cards from storage and update the dashboard accordingly.
+        savedCards = await loadDashboardCards() ?? []
+        observeValuesForDashboardCards()
```

Comment:
❓  I see that we are starting the observation in view model's `init`. Do we have to do it again here?

Reply:
I wanted to trigger updating dashboard cards following the cards saved from storage in this case. Updated the logic in 3debc967f077bc440c137e72b275c26c352f586e to trigger the update directly instead.